torchsde/_core/base_sde.py

```diff
@@ -114,10 +70,120 @@ def gdg_prod(self, t, y, v):
         pass
 
 
-class SDEStratonovich(BaseSDE):
+class ForwardSDE(BaseSDE):
 
-    def __init__(self, noise_type):
-        super(SDEStratonovich, self).__init__(noise_type=noise_type, sde_type='stratonovich')
+    def __init__(self, base_sde):
+        super(ForwardSDE, self).__init__(sde_type=base_sde.sde_type, noise_type=base_sde.noise_type)
+        self._base_sde = base_sde
+
+        # Register the core function. This avoids polluting the codebase with if-statements.
+        self.g_prod = {
+            settings.NOISE_TYPES.diagonal: self.g_prod_diagonal,
+            settings.NOISE_TYPES.additive: self.g_prod_additive,
+            settings.NOISE_TYPES.scalar: self.g_prod_scalar,
+            settings.NOISE_TYPES.general: self.g_prod_general
+        }[base_sde.noise_type]
+        self.gdg_prod = {
+            settings.NOISE_TYPES.diagonal: self.gdg_prod_diagonal,
+            settings.NOISE_TYPES.additive: self._skip,
```

Comment:
Done.

Reply:
Let's do this just for the "globals" if possible. I'm trying to follow the google style on this, which recommends importing modules as much as possible, since it's fairly possible to have multiple functions with the same name in different modules.